src/MultiprocessingEngine.py

```diff
@@ -19,40 +19,43 @@
 #
 from __future__ import print_function
 import multiprocessing
+import multiprocessing.pool
 #
 import crosscat.LocalEngine as LE
 import crosscat.utils.sample_utils as su
 
+class Pool(multiprocessing.pool.Pool):
 
-class MultiprocessingEngine(LE.LocalEngine):
-    """A simple interface to the Cython-wrapped C++ engine
-
-    MultiprocessingEngine holds no state other than a seed generator.
-    Methods use resources on the local machine.
-
-    """
-
-    def __init__(self, seed=None, cpu_count=None):
-        """Initialize a MultiprocessingEngine
+    def __del__(self):
+        self.terminate()
 
-        This is really just setting the initial seed to be used for
-        initializing CrossCat states.  Seeds are generated sequentially
+class MultiprocessingEngine(LE.LocalEngine):
+    """A simple interface to the Cython-wrapped C++ engine"""
 
-        """
+    def __init__(self, seed=None, pool=None, cpu_count=None):
         super(MultiprocessingEngine, self).__init__(seed=seed)
-        self.pool = multiprocessing.Pool(cpu_count)
+        if cpu_count is None:
+            if pool is None:
+                raise ValueError('Specify a cpu count or a process pool')
+        else:
+            pool = Pool(cpu_count)
+        self.pool = pool
         self.mapper = self.pool.map
         return
-    
+
     def __enter__(self):
         return self
 
-    def __del__(self):
```

Comment:
You expect that this is un-needed because the pool will clean up after itself anyway on its own garbage collection? That does seem to be the case, but then I wonder why someone put this here...